browser/plugins/three_text_geometry.plugin.js

```diff
@@ -107,20 +117,61 @@
 		this.geometry.computeFaceNormals()
 
 		this.dirty = false
+
+		this.updated = true
 	}
 
-	ThreeTextGeometry.prototype.state_changed = function(ui) {
-		if (ui) {
+	ThreeTextGeometry.prototype.update_state = function() {
+		if (!this.font || !this.font.font || (this.state.fontId !== this.font.font.id)) {
+			if (!this.loadPromise) {
+				this.font = this.fontSelector.getById(this.state.fontId)
+
+				if (!this.font.font) {
+					var that = this
+
+					var dfd = when.defer()
+
+					var doLoad = function() {
+						that.fontSelector.fontLoader.load(that.font.url, function(font) {
+							that.font.font = font
+							that.reconstructGeometry()
+
+							dfd.resolve()
+						})
+					}
+
+					if (this.loadPromise) {
```

Comment:
there is no way loadPromise is set here if it wasn't on line 126. I would reformat this block by reversing the logic of the if's and getting early-out, unless we really need to load